test/document.test.js

```diff
@@ -2816,12 +2820,16 @@ describe('document', function() {
 
       var MyModel = db.model('gh4014', schema);
 
-      MyModel.
-        where('geo').near({ center: [50, 50] }).
-        exec(function(error) {
-          assert.ifError(error);
-          done();
-        });
+      MyModel.on('index', function(err) {
```

Comment:
In general its better to use `MyModel.init().then(() => {})` instead of `MyModel.on('index')`, because with the latter if you register the event listener after the index build is finished, you'll never get a callback.

Reply:
Yes it really makes sense! There are similar test codes like this elsewhere and I will try to update it in future fix to make the test more robust.

Although index takes time to build, if we use some sort of index cache in mongoose in the future, or `MyModel.on('index')` is not placed right after `MyModel.model()` in the same event loop, that's a potential issue.